### DISCUSSION TITLE
Add a user and continue recipes using it

### DISCUSSION THREAD

**Discussion by mihaigalos** [casey/just#1268] [Q&A] [answered]
Hi,

On Linux, I would like to add a user and continue the recipes using it. Is something like this possible without exiting the initial `just` invocation?

```just
add_user user:
    sudo adduser {{ user }}

run user:
    just add {{ user }}
    # sudo su {{ user }} here, or similar ..
    just setup_zsh # Setup zsh for that user, instead of the user who initiated the just invocation.
```

**Comment by casey** [comment, marked as answer]
Can you use a dependency?

```
add user:
    sudo adduser {{ user }}

run user: (add user)
    # sudo su {{ user }} here, or similar ..
    # Setup zsh for that user, instead of the user who initiated the just invocation.
```

**Reply by mihaigalos** [reply]
Yes, no constraints on the solution.
The problem remains, however, that just process doesn't continue with the current user.

My temporary solution is this, maybe we can improve it further?

```bash
add user:
    sudo adduser {{ user }}

run user: (add user)
    sudo -u {{ user }} -- sh -c '\
        just setup_zsh; \
        just remove_temporaries; \
    '
```

**Reply by casey** [reply]
How about this:

```
# set positional arguments to avoid problems with splitting and quoting
set positional-arguments

add user:
    sudo adduser {{ user }}

sudo user +args: (add user)
    sudo -u {{ user }} just "${@:2}"
```

Then to add user `foo` and run recipe `bar`:

```
$ just sudo foo bar
```

Using `+args` allows passing multiple recipes.

**Reply by mihaigalos** [reply]
> sudo -u {{ user }|} just "${@:2}"

I don't understand the pipe (`|`) in the user. I suspect it's a typo?

Otherwise, this looks beautiful!

**Reply by casey** [reply]
Oops, yeah, typo!
